makefiles/stdio.inc.mk

```diff
@@ -0,0 +1,64 @@
+STDIO_MODULES := \
+  stdio_cdc_acm \
+  stdio_null \
+  stdio_uart \
+  slipdev_stdio \
+  stdio_ethos \
+  stdio_rtt \
+  #
+
+# Since USEMODULE and DEFAULT_MODULEs are recursively expanded we need to
+# disable DEFAULT_MODULEs before any of there dependencies are evaluated,
+# otherwise the disabled MODULE will be in USEMODULE (triggering) its
+# dependencies, and then removed but leaving its dependencies
+ifneq (,$(filter $(filter-out stdio_rtt,$(STDIO_MODULES)),$(USEMODULE)))
+  # stdio_rtt cannot be used when another STDIO is loaded
+  DISABLE_MODULE += stdio_rtt
+endif
+
+ifneq (,$(filter $(filter-out stdio_dc_acm,$(STDIO_MODULES)),$(USEMODULE)))
+  # stdio_cdc_acm cannot be used when another STDIO is loaded
+  DISABLE_MODULE += stdio_cdc_acm
+endif
+
+ifneq (,$(filter newlib,$(USEMODULE)))
+  ifeq (,$(filter $(STDIO_MODULES),$(USEMODULE)))
+    USEMODULE += stdio_uart
+  endif
+endif
```

Comment:
Adapted from `Makefile.dep`, I know, but does this dependency even make sense? Why enforce a STDIO when `newlib` is there? AFAIK `newlib` should just work fine without STDIO, or am I mistaken?

Reply:
> Adapted from Makefile.dep, I know, but does this dependency even make sense? Why enforce a STDIO when newlib is there? AFAIK newlib should just work fine without STDIO, or am I mistaken?

This might just be a way of saying if `newlib` is included then default `stdio` is `stdio_uart`.

Reply:
Yea, but basically this is why `stdio_null` is necessary, right? Why does there need to be a `stdio` at all?

Reply:
> Yea, but basically this is why stdio_null is necessary, right? Why does there need to be a stdio at all?

It doesn't but most users expect to see something if they do `make term` so it makes sense to usually include it.

Maybe more can be done towards cleaning this up by actually using `DEFAULT_MODULE` here, but I would rather leave it for later.

Reply:
Yeahyeah, this was just a thought, not actually requesting a solution here.